### DISCUSSION TITLE
Compiling error

### DISCUSSION THREAD

**Discussion by Ponpont** [grblHAL/core#106] [Q&A] [answered]
Hi guys,

I am currently using the old preconfiguration and already compiled FW on my grblHAL Breakout Board with Teensy 4.1 on it.

I have made some upgrade on my machine (limit switch for auto square gantry, RGB light for the rgb plugin, etc) and I want to update the firmware. I've made all the changes needed in the various files while following the wiki compiling instruction but when I tried to compile everything, I got this fatal error. I'am using the Arduino IDE with teensyduino added

`C: lUsers \CNC-01 \ Documents \Arduino\libraries\grblMAL_Teensy4\src\networking\networking.h:48:22: fatal error:
compilation terminated.
Erreur
de compilation pour la carte Teensy 4.1
lwipopts.h: No such
or dIrectory`

Have you guys already face this one ?

Cheers and keep up the good work, my CNC works flawlessly from the start

**Comment by terjeio** [comment, marked as answer]
It is a dependency that should only be compiled in when ethernet is enabled. A workaround is to delete _networking/networking.c_ or comment out the code by adding  ` /*`  as the first line and  ` */` as the last. I'll fix this in the next commit.

**Reply by Ponpont** [reply]
Well that was simple, thank you !

I just read the readme.md that state to add the library for networking (RTFM duh)

One thing that is still not clear is how to use auto squared function on ganged axis. I'm currently using B motor as my Y ganged motor. I have 2 limit switches on Y and B min. Auto square is performed when I issue the homing cycle ?

Thank you, for your help !

**Reply by terjeio** [reply]
> Auto square is performed when I issue the homing cycle ?

Yes, but you have to enable the axis as auto squared, not only ganged, in _my_machine.h_. Check wiring before homing, triggering either limit switch should show the Y axis limit as active. Use the `$pins` command to check which pin is assigned for limit inputs.

There are a few $-settings available for tuning auto squaring, e.g. `$171` can be used to set an offset if the switches are not perfectly aligned.

**Reply by Ponpont** [reply]
Well you just raise awareness on one issuse I was having with Y axis limit switch !
It was not B axis I had wired but A axis, so it needs A limit switch to be wired and also need to remove the jumper pad soldered from A ganged to Y back to the normal state A to A

$171 is awesome I didn't understand fully what it meant at first

CNC is back working 100%, thank you !
